### DISCUSSION TITLE
[AUTH] Hide codes on web

### DISCUSSION THREAD

**Discussion by Vaisakhkm2625** [ente-io/ente#4239] [Enhancements]
please provide a feature to hide TOTP Codes on web...

implementation seems really simple, that's why asking, something like

```js
document.querySelectorAll("#__next > div:nth-child(2) > div:nth-child(4) > div > div:nth-child(1) > div:nth-child(1) > div:nth-child(2) > div:nth-child(1) > p:nth-child(3)").forEach((el)=>{el.innerText="******"}) 
```
![image](https://github.com/user-attachments/assets/70da666d-776b-4be4-80fd-35b56a6b1d29)

![image](https://github.com/user-attachments/assets/a86703c3-0a71-4b4c-872c-3d3a1fc28ae5)

**Comment by Vaisakhkm2625** [comment]
i gone through the code, and i understand that this is not a easy fix if needed to be done in the proper way, like having preference settings on web...

https://github.com/ente-io/ente/blob/74117db8b059358dd6854138062d502fdc877a3e/auth/lib/ui/code_widget.dart#L62

it is accessing `preference_service`
```
auth/lib/services/preference_service.dart
```
which is not available in the web, which also limits the features like pinning...

# Temp workaround
so for now, workaround

```
document.querySelectorAll("#__next > div:nth-child(2) > div:nth-child(4) > div> div > div > div:nth-child(2) > div:nth-child(3) > p:nth-child(2)").forEach((el)=>{el.style.color="#00000000";el.style.backgroundColor="#000000";});

document.querySelectorAll("#__next > div:nth-child(2) > div:nth-child(4) > div > div:nth-child(1) > div:nth-child(1) > div:nth-child(2) > div:nth-child(1) > p:nth-child(3)").forEach((el)=>{el.style.color="#00000000";el.style.backgroundColor="#000000";});

```
as auto injected script in my browser as violentmonkeyscript

```js

(function() {
    'use strict';

    // Add custom styles directly
    const style = document.createElement('style');
    style.textContent = `

        #__next > div:nth-child(2) > div:nth-child(4) > div > div > div > div:nth-child(2) > div:nth-child(3) > p:nth-child(2) {
            color: #00000000 !important;
            background-color: #000000 !important;
        }
        #__next > div:nth-child(2) > div:nth-child(4) > div > div:nth-child(1) > div:nth-child(1) > div:nth-child(2) > div:nth-child(1) > p:nth-child(3) {
            color: #00000000 !important;
            background-color: #000000 !important;
        }

        #__next > div:nth-child(2) > div:nth-child(4) > div > div > div > div:nth-child(2) > div:nth-child(3) > p:nth-child(2):hover {
            color: white !important; /* Change text color on hover */
            background-color: black !important; /* Change background color on hover */
        }


        #__next > div:nth-child(2) > div:nth-child(4) > div > div:nth-child(1) > div:nth-child(1) > div:nth-child(2) > div:nth-child(1) > p:nth-child(3):hover {
            color: white !important; /* Change text color on hover */
            background-color: black !important; /* Change background color on hover */
        }
    `;
    document.head.appendChild(style);
})();



```

**Comment by Vaisakhkm2625** [comment]
Now only i found that i where searching in wrong folder, i though webapp is generated from flutter,
But today only i noticed that it's in next js.
https://github.com/ente-io/ente/blob/main/web/apps/auth/src/pages/auth.tsx

**Reply by mnvr** [reply]
We'll not be accepting pull requests to add this feature right now. It is great that you've found a way to make it work for yourself already, I'm just mentioning this to save you effort in opening a pull request that'd not be accepted.

Depending on future traction / upvotes on this feature request, we can revisit this in the future.

**Reply by Vaisakhkm2625** [reply]
Thanks for the reply, i understood that, hence i haven't started writing it..
